Makefile

```diff
@@ -3,35 +3,69 @@ PROJECT          ?= ansibleplaybookbundle
 TAG              ?= latest
 BROKER_IMAGE     = $(REGISTRY)/$(PROJECT)/ansible-service-broker
 BUILD_DIR        = "${GOPATH}/src/github.com/openshift/ansible-service-broker/build"
+INSTALL_DIR      ?= /usr/local
 
-install: $(shell find cmd pkg)
-	go install -ldflags="-s -w" ./cmd/broker
+build: vendor
+	go build -ldflags="-s -w" ./cmd/broker
 
-${GOPATH}/bin/mock-registry: $(shell find cmd/mock-registry)
-	go install ./cmd/mock-registry
+#install, run, and uninstall are geared towards running on your local host
+install: build
+	cp broker ${INSTALL_DIR}/bin/ansible-service-broker
+	mkdir -p ${INSTALL_DIR}/etc/ansible-service-broker
+	cp etc/ex.dev.config.yaml ${INSTALL_DIR}/etc/ansible-service-broker/config.yaml
 
-# Will default run to dev profile
-run: install vendor
-	@${GOPATH}/src/github.com/openshift/ansible-service-broker/scripts/runbroker.sh dev
+run: install 
+	${INSTALL_DIR}/bin/ansible-service-broker --config ${INSTALL_DIR}/etc/ansible-service-broker/config.yaml
 
-deploy:
-	@${GOPATH}/src/github.com/openshift/ansible-service-broker/scripts/deploy.sh
+uninstall:
+	rm  -f ${INSTALL_DIR}/bin/ansible-service-broker
+	rm -rf ${INSTALL_DIR}/etc/ansible-service-broker
 
-run-mock-registry: ${GOPATH}/bin/mock-registry vendor
-	@${GOPATH}/src/github.com/openshift/ansible-service-broker/cmd/mock-registry/run.sh
+#build-image will build a docker image using the source build of the broker
+prepare-build-image: build
+	cp broker build/broker
 
-prepare-build: install
-	cp "${GOPATH}"/bin/broker build/broker
+build-image: prepare-build-image
+	docker build -f ${BUILD_DIR}/Dockerfile-src ${BUILD_DIR} -t ${BROKER_IMAGE}:${TAG}
+	@echo
+	@echo "Remember you need to push your image before calling make deploy"
+	@echo "    docker push ${BROKER_IMAGE}:${TAG}"
 
-build: prepare-build
+#release will build an image from prebuilt rpms, push will push the image to docker.io
+#Use release instead of release-image. We may add push to it later, but for now they are discrete.
+release-image:
 	docker build ${BUILD_DIR} -t ${BROKER_IMAGE}:${TAG}
 	@echo
 	@echo "Remember you need to push your image before calling make deploy"
-	@echo "    docker push ${BROKER_IMAGE}:${TAG}"
+	@echo "    make push"
+
+push:
+	docker push ${BROKER_IMAGE}:${TAG}
+
+release: release-image
```

Comment:
@jmrodri I think the logic was that if we do that and later wanted release to do release & push we wouldn't need to change the workflow you would still just run release in the future and not have to change your workflow. 

Granted we could just put the docker push command under release at that point, but this would also allows them to be run as separate steps if you want to check the image in between steps if you desired.

I think I was tarcking @jmrodi's advice but if I did it wrong/too convoluted happy to fix.